lib/SILOptimizer/Analysis/ArraySemantic.cpp

```diff
@@ -682,3 +684,36 @@ bool swift::ArraySemanticsCall::replaceByValue(SILValue V) {
   removeCall();
   return true;
 }
+
+bool swift::ArraySemanticsCall::replaceByAppendingValues(SILModule &M, SILFunction* appendFn, llvm::SmallVectorImpl<SILValue> &Vs, ArrayRef<ProtocolConformanceRef> &conformances) {
+  assert(getKind() == ArrayCallKind::kAppendContentsOf &&
+         "Must be an append_contentsOf call");
+  assert(appendFn && "Must provide an append SILFunction");
+
+  // We only handle loadable types.
+  for(auto& V : Vs) {
+    if (!V->getType().isLoadable(SemanticsCall->getModule()))
```

Comment:
SemanticsCall->getModule() and SemanticsCall->getLoc() should be stored in locals for readability.